sdk/helper/certutil/helpers.go

```diff
@@ -2150,3 +2156,101 @@ func IsPSS(algorithm x509.SignatureAlgorithm) bool {
+	CRLIssuer         asn1.RawValue         `asn1:"optional,tag:2"`
+}
+type distributionPointName struct {
+	FullName     []asn1.RawValue  `asn1:"optional,tag:0"`
+	RelativeName pkix.RDNSequence `asn1:"optional,tag:1"`
+}
+
+func CreateDeltaCRLExtension(deltaUrls []string) (pkix.Extension, error) {
+	var deltaDistributionPoints []deltaDistributionPoint
+	for _, deltaUrl := range deltaUrls {
+		delta := deltaDistributionPoint{
+			DistributionPoint: distributionPointName{
+				FullName: []asn1.RawValue{
+					{
+						Tag:   6,
+						Class: 2,
+						Bytes: []byte(deltaUrl),
+					},
+				},
+			},
+		}
+		deltaDistributionPoints = append(deltaDistributionPoints, delta)
+	}
+	asn1Value, err := asn1.Marshal(deltaDistributionPoints)
+	if err != nil {
+		return pkix.Extension{}, err
+	}
+	return pkix.Extension{
+		Id:       FreshestCrlOid,
+		Value:    asn1Value,
+		Critical: false,
+	}, nil
+}
+
+// Adapted From: https://cs.opensource.google/go/go/+/master:src/crypto/x509/parser.go?q=CRLDistributionPoints&ss=go%2Fgo
+func ParseDeltaCRLExtension(certificate *x509.Certificate) ([]string, error) {
+	logCert := base64.StdEncoding.EncodeToString(certificate.Raw)
+	urls := []string{}
+	for _, extension := range certificate.Extensions {
+		if extension.Id.Equal(FreshestCrlOid) {
+			val := cryptobyte.String(extension.Value)
+			answer := !val.ReadASN1(&val, asn1.TagSequence|32)
+			if answer { // classConstructed SEQUENCE
+				return urls, errors.New("x509: invalid CRL distribution points")
+			}
+			for !val.Empty() {
+				var dpDER cryptobyte.String
+				if !val.ReadASN1(&dpDER, asn1.TagSequence|32) {
+					return urls, errors.New("x509: invalid CRL distribution point")
+				}
+				var dpNameDER cryptobyte.String
+				var dpNamePresent bool
+				if !dpDER.ReadOptionalASN1(&dpNameDER, &dpNamePresent, asn1.ClassUniversal|32|128) {
+					return urls, errors.New("x509: invalid CRL distribution point")
+				}
+				if !dpNamePresent {
+					continue
+				}
+				if !dpNameDER.ReadASN1(&dpNameDER, asn1.ClassUniversal|32|128) {
+					return urls, errors.New("x509: invalid CRL distribution point")
+				}
+				for !dpNameDER.Empty() {
+					if !dpNameDER.PeekASN1Tag(asn1.TagOID | 128) { // Context is GeneralName
+						break
+					}
+					var uri cryptobyte.String
+					if !dpNameDER.ReadASN1(&uri, asn1.TagOID|128) { // Context is GeneralName
+						return urls, errors.New("x509: invalid CRL distribution point")
+					}
+					urls = append(urls, string(uri))
+				}
+			}
+			return urls, nil
+		}
+	}
+	fmt.Print(logCert)
+	return urls, nil
+}
+
+func AddDeltaCRLExtension(data *CreationBundle, certTemplate *x509.Certificate) {
+	if data.Params != nil && data.Params.URLs != nil && data.Params.URLs.DeltaCRLDistributionPoints != nil &&
+		len(data.Params.URLs.DeltaCRLDistributionPoints) > 0 {
+		extension, err := CreateDeltaCRLExtension(data.Params.URLs.DeltaCRLDistributionPoints)
+		if err != nil {
+			return
```

Comment:
Do we really want to hide the error here? Should we not at least return a warning?

Reply:
You are right hiding an error here is not a good idea, and might lead to a lot of confusion down the road.  I don't think that this ever could/should error, and because of placement, it would take a lot of changes to be able to return a warning from here - I've changed this to just return an error (fail fast), if that's okay with you?

Reply:
Yup returning an error here is okay.